modules/rgbd/src/large_kinfu.cpp

```diff
@@ -0,0 +1,361 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+// This code is also subject to the license terms in the LICENSE_KinectFusion.md file found in this
+// module's directory
+
+#include "fast_icp.hpp"
+#include "hash_tsdf.hpp"
+#include "kinfu_frame.hpp"
+#include "precomp.hpp"
+#include "submap.hpp"
+#include "tsdf.hpp"
+
+namespace cv
+{
+namespace large_kinfu
+{
+using namespace kinfu;
+
+Ptr<Params> Params::defaultParams()
+{
+    Params p;
+
+    //! Frame parameters
+    {
+        p.frameSize = Size(640, 480);
+
+        float fx, fy, cx, cy;
+        fx = fy = 525.f;
+        cx      = p.frameSize.width / 2.0f - 0.5f;
+        cy      = p.frameSize.height / 2.0f - 0.5f;
+        p.intr  = Matx33f(fx, 0, cx, 0, fy, cy, 0, 0, 1);
+
+        // 5000 for the 16-bit PNG files
+        // 1 for the 32-bit float images in the ROS bag files
+        p.depthFactor = 5000;
+
+        // sigma_depth is scaled by depthFactor when calling bilateral filter
+        p.bilateral_sigma_depth   = 0.04f;  // meter
+        p.bilateral_sigma_spatial = 4.5;    // pixels
+        p.bilateral_kernel_size   = 7;      // pixels
+        p.truncateThreshold       = 0.f;    // meters
+    }
+    //! ICP parameters
+    {
+        p.icpAngleThresh = (float)(30. * CV_PI / 180.);  // radians
+        p.icpDistThresh  = 0.1f;                         // meters
+
+        p.icpIterations = { 10, 5, 4 };
+        p.pyramidLevels = (int)p.icpIterations.size();
+    }
+    //! Volume parameters
+    {
+        float volumeSize                   = 3.0f;
+        p.volumeParams.type                = VolumeType::TSDF;
+        p.volumeParams.resolution          = Vec3i::all(512);
+        p.volumeParams.pose                = Affine3f().translate(Vec3f(-volumeSize / 2.f, -volumeSize / 2.f, 0.5f));
+        p.volumeParams.voxelSize           = volumeSize / 512.f;            // meters
+        p.volumeParams.tsdfTruncDist       = 7 * p.volumeParams.voxelSize;  // about 0.04f in meters
+        p.volumeParams.maxWeight           = 64;                            // frames
+        p.volumeParams.raycastStepFactor   = 0.25f;                         // in voxel sizes
+        p.volumeParams.depthTruncThreshold = p.truncateThreshold;
```

Comment:
Does this thing duplicate the params in `volume.cpp`?

Reply:
Yes, pretty much. :/